### ISSUE TITLE
[Chapter 14] Cache tag helper doesn't work

### ISSUE THREAD

**Issue by oleg-glushko** [mikebrind/Razor-Pages-In-Action#2]
I can't get the [CacheTagHelper](https://github.com/mikebrind/Razor-Pages-In-Action/tree/main/Chapter14/CacheTagHelper) example to work. It always returns a literal `<vc:cities />` instead of a rendered component. But an app displays cities as soon as I change this tag helper with `@await Component.InvokeAsync("Cities")`.

I'm running it on Windows 10 x64 with latest patches, VS2022 Community Edition, .NET 6.0.14

**Comment by mikebrind** [comment]
Did you remember to [activate tag helpers](https://github.com/mikebrind/Razor-Pages-In-Action/blob/85205cbc63550b6475f52a286aa37c1009bea62b/Chapter14/CacheTagHelper/CityBreaks/Pages/_ViewImports.cshtml#L5) for your application?

**Comment by oleg-glushko** [comment]
I "did" - it's already activated in this example. I also tried to add this directive to Index.cshtml, still no luck.

**Comment by mikebrind** [comment]
Did you try downloading the example here and running it?

**Comment by oleg-glushko** [comment]
Of course, I'm trying to run the example linked in the [first post](https://github.com/mikebrind/Razor-Pages-In-Action/issues/2#issue-1588746837). But this behavior also reproduces in another simple test project.

**Comment by mikebrind** [comment]
OK. This was a total mystery, but all of a sudden this morning, I got the same issue in other projects. I had installed newer versions of the .NET 7 SDKs and view component tag helpers stopped working. In my case, I added a _global.json_ pinning the SDK version to 6.0.308, and the tag helpers started working again.

I'm guessing there is an issue with one of the .NET 7 SDKs, which may or may not be resolved in the latest one. I haven't checked. 

Execute `dotnet --list-sdks` from a command prompt to see which SDKs you have for .NET 6 and add a _global.json_ file to the solution folder pinning the version to one that you have:

```
{
    "sdk": {
        "version": "6.0.308"
    }
}
```

Ref: https://www.learnrazorpages.com/publishing/publish-to-iis#fixing-the-sdk-version-used-for-publishing

**Comment by oleg-glushko** [comment]
It did the trick after I pinpointed a version of the installed .NET 6 SDK. Also, yesterday I stumbled upon [this issue](https://github.com/dotnet/aspnetcore/issues/46740). It seems like your guess is correct - there's a regression in a recent .NET 7 SDK. It did the trick after I pinpointed a version of the installed .NET 6 SDK. Also, yesterday I stumbled upon this issue. It seems like your guess is correct - there's a regression in a recent .NET 7 SDK. Thank you, now it's time to finish the last chapter of your neat book. I have a couple more not-so-critical issues, but I'll post them separately later, along with solutions.
